docker/build.sh

```diff
@@ -104,6 +104,31 @@ gquant_ver=$(grep version gQuant/setup.py | sed "s/^.*version='\([^;]*\)'.*/\1/"
 CONTAINER="nvidia/cuda:${CUDA_STR}-runtime-${OS_STR}"
 D_CONT=${D_CONT:="gquant/gquant:${gquant_ver}-${CUDA_STR}_${OS_STR}_${RAPIDS_VERSION}_${MODE_STR}"}
 
+
+cat << 'EOF' > sacrebleu.patch
+--- nemo/collections/nlp/metrics/sacrebleu.py
++++ sacrebleu_fix.py
+@@ -61,13 +61,16 @@
+ VERSION = '1.3.5'
+ 
+ try:
++    import threading
+     # SIGPIPE is not available on Windows machines, throwing an exception.
+     from signal import SIGPIPE
+ 
+     # If SIGPIPE is available, change behaviour to default instead of ignore.
+     from signal import signal, SIG_DFL
+ 
+-    signal(SIGPIPE, SIG_DFL)
++
++    if threading.current_thread() == threading.main_thread():
++        signal(SIGPIPE, SIG_DFL)
+ 
+ except ImportError:
+     logging.warning('Could not import signal.SIGPIPE (this is expected on Windows machines)')
+EOF
+
```

Comment:
Why do we need this patch? Is NeMo broken?

Reply:
When running ray tune without the patch, that code would raise a ValueError.
```
  File "/home/quant/NeMo/nemo/collections/nlp/metrics/sacrebleu.py", line 73, in <module>
    signal(SIGPIPE, SIG_DFL)
  File "/home/quant/miniconda3/lib/python3.7/signal.py", line 47, in signal
    handler = _signal.signal(_enum_to_int(signalnum), _enum_to_int(handler))
ValueError: signal only works in main thread
```

The fix was discussed here:
https://bugs.python.org/issue38904
https://bugs.python.org/issue39042

I'm not sure why this worked before. By refactoring nemo stuff as a separate module and ray doing its magic with workers and tasks, somehow that signal is not being imported in main thread anymore.